final class kotlinx.coroutines.internal.ExceptionsConstructorKt$safeCtor$1 extends kotlin.jvm.internal.Lambda implements kotlin.jvm.functions.Function1
{
    final kotlin.jvm.functions.Function1 $block;

    void <init>(kotlin.jvm.functions.Function1)
    {
        kotlin.jvm.functions.Function1 $r1;
        kotlinx.coroutines.internal.ExceptionsConstructorKt$safeCtor$1 r0;

        r0 := @this: kotlinx.coroutines.internal.ExceptionsConstructorKt$safeCtor$1;

        $r1 := @parameter0: kotlin.jvm.functions.Function1;

        r0.<kotlinx.coroutines.internal.ExceptionsConstructorKt$safeCtor$1: kotlin.jvm.functions.Function1 $block> = $r1;

        specialinvoke r0.<kotlin.jvm.internal.Lambda: void <init>(int)>(1);

        return;
    }

    public volatile java.lang.Object invoke(java.lang.Object)
    {
        java.lang.Throwable $r2;
        java.lang.Object $r1;
        kotlinx.coroutines.internal.ExceptionsConstructorKt$safeCtor$1 r0;

        r0 := @this: kotlinx.coroutines.internal.ExceptionsConstructorKt$safeCtor$1;

        $r1 := @parameter0: java.lang.Object;

        $r2 = (java.lang.Throwable) $r1;

        $r2 = virtualinvoke r0.<kotlinx.coroutines.internal.ExceptionsConstructorKt$safeCtor$1: java.lang.Throwable invoke(java.lang.Throwable)>($r2);

        return $r2;
    }

    public final java.lang.Throwable invoke(java.lang.Throwable)
    {
        java.lang.Throwable $r1, r6;
        kotlin.Result$Companion $r4;
        kotlin.jvm.functions.Function1 r2;
        kotlinx.coroutines.internal.ExceptionsConstructorKt$safeCtor$1 r0;
        java.lang.Object $r3, $r5;
        java.lang.String $r7, $r8;
        boolean $z0;

        r0 := @this: kotlinx.coroutines.internal.ExceptionsConstructorKt$safeCtor$1;

        $r1 := @parameter0: java.lang.Throwable;

        r2 = r0.<kotlinx.coroutines.internal.ExceptionsConstructorKt$safeCtor$1: kotlin.jvm.functions.Function1 $block>;

        $r3 = null;

     label01:
        $r4 = <kotlin.Result: kotlin.Result$Companion Companion>;

        $r5 = interfaceinvoke r2.<kotlin.jvm.functions.Function1: java.lang.Object invoke(java.lang.Object)>($r1);

        r6 = (java.lang.Throwable) $r5;

        $r7 = virtualinvoke $r1.<java.lang.Throwable: java.lang.String getMessage()>();

        $r8 = virtualinvoke r6.<java.lang.Throwable: java.lang.String getMessage()>();

        $z0 = staticinvoke <kotlin.jvm.internal.Intrinsics: boolean areEqual(java.lang.Object,java.lang.Object)>($r7, $r8);

     label02:
        if $z0 != 0 goto label05;

     label03:
        $r7 = virtualinvoke r6.<java.lang.Throwable: java.lang.String getMessage()>();

        $r8 = virtualinvoke $r1.<java.lang.Throwable: java.lang.String toString()>();

        $z0 = staticinvoke <kotlin.jvm.internal.Intrinsics: boolean areEqual(java.lang.Object,java.lang.Object)>($r7, $r8);

     label04:
        if $z0 != 0 goto label05;

        $r1 = null;

        goto label06;

     label05:
        $r1 = r6;

     label06:
        $r5 = staticinvoke <kotlin.Result: java.lang.Object constructor-impl(java.lang.Object)>($r1);

     label07:
        goto label09;

     label08:
        r6 := @caughtexception;

        $r4 = <kotlin.Result: kotlin.Result$Companion Companion>;

        $r5 = staticinvoke <kotlin.ResultKt: java.lang.Object createFailure(java.lang.Throwable)>(r6);

        $r5 = staticinvoke <kotlin.Result: java.lang.Object constructor-impl(java.lang.Object)>($r5);

     label09:
        $z0 = staticinvoke <kotlin.Result: boolean isFailure-impl(java.lang.Object)>($r5);

        if $z0 == 0 goto label10;

        goto label11;

     label10:
        $r3 = $r5;

     label11:
        $r1 = (java.lang.Throwable) $r3;

        return $r1;

        catch java.lang.Throwable from label01 to label02 with label08;
        catch java.lang.Throwable from label03 to label04 with label08;
        catch java.lang.Throwable from label06 to label07 with label08;
    }
}
